### DISCUSSION TITLE
Issues point mass navigation gyro

### DISCUSSION THREAD

**Discussion by MaevaGuerrier** [PKU-Alignment/safety-gymnasium#120] [Q&A]
Hello, I need to use the gyro to obtain the angular velocity. I found out that if I keep my agent with a force of 0 and apply an orientation of .1 (which implies that the agent is staying at the same position and rotating in the same place) the gyro should also give me an angular velocity of .1, correct ? However, this is not the case 
 Here is the action I am feeding to the agent action [0.  0.1]

Here is the observation space: 

```python
[ 3.76420450e-03 -1.25403920e+01  9.81000000e+00  7.34045467e-06
  9.15165221e-03  0.00000000e+00  0.00000000e+00  0.00000000e+00
 -4.42776151e-01 -4.93031965e-01 -8.31834237e-02  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  1.31161504e-01
  1.90387407e-01  5.92259027e-02  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  7.67086504e-02
  8.05358533e-01  7.28649882e-01  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00
  0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]

```
My gyro value is  -4.42776151e-01 which should have been .1, is there something I am missing ? Any help is appreciated, thank you

**Comment by muchvo** [comment]
Sorry for missing this message earlier. How are you ensuring the rotation at an angular velocity of 0.1? Could you please share a minimal reproducible script so I can better understand and replicate the issue?

**Reply by MaevaGuerrier** [reply]
Thank you for your reply. 
Actually what I need is to retrieve the orientation theta of the point agent for my code and was trying to check the gyro to see if everything was making sense. 
I see that If I use the env.task.agent.rot it does not give me a proper value or sometime None. I can see that in base task we have but the values is not changing over time as I give random action to the agent

 ```python
        if self.agent.rot is None:
            world_config['agent_rot'] = self.random_generator.random_rot()
        else:
            world_config['agent_rot'] = float(self.agent.rot)
```

Right now I am using the quaternion to retrieve my orientation but somehow my code is not working properly I  tested the behavior in Octave/Matlab so I wonder if the way I try to get the point agent orientation is wrong. 

I saw that the angular velocity around the z axis is actually in m/s in the doc but when I change my omega into m/s those values the behaviour is getting further away from what I expect. 

I am trying to understand the mapping between omega and mujoco.

```python

import safety_gymnasium
import numpy as np
import matplotlib.pyplot as plt
from scipy.integrate import solve_ivp
from unicycle_cbf_rl import step_unicycle_cbf
from time import sleep
import os
import mujoco
import mujoco_py
from scipy.spatial.transform import Rotation as R


env = safety_gymnasium.make('SafetyPointGoal1-v0', render_mode='human')

obs, info = env.reset()

while True:


    rotation = R.from_quat(env.task.data.xquat[1]) # quat matrix extracted to get only agent info
    euler = rotation.as_euler('xyz', degrees=False)
    theta = euler[0] # yaw angle
    print(f"theta {theta}")

    omega = .1
    action = np.array([0, omega]) 
    print(f"gyro value {obs[8]}")

    obs, reward, terminated, truncated, info = env.step(action) 

    if terminated or truncated: 
        obs, info = env.reset()

    env.close()




```


Thank you again for your reply.
Please, let me know if something need more details.

**Comment by muchvo** [comment]
Apologies for the delayed response. I ran your code, and this behavior is expected. In MuJoCo, control of speed and position is essentially implemented through the creation of integrated servos. Considering factors like friction, the actual stable values may not exactly match the set values. For more details, please refer to [this link](https://mujoco.readthedocs.io/en/latest/XMLreference.html#actuator-velocity).

**Reply by MaevaGuerrier** [reply]
No worries, thank you for the answer!
